modules/xray/variables.tf

```diff
@@ -0,0 +1,126 @@
+  stage            = "${var.stage}"
+  service_name     = "${var.service_name}"
+  tf_service_name  = "${var.tf_service_name != "" ? var.tf_service_name : "tf-${var.service_name}"}"
+  sls_service_name = "${var.sls_service_name != "" ? var.sls_service_name : "sls-${var.service_name}"}"
+
+  tags = "${map(
+    "Service", "${var.service_name}",
+    "Stage", "${var.stage}",
+  )}"
+}
+
+# Capture repeated/complicated AWS IAM resources to a single location.
+locals {
+  # Serverless CloudFormation stack ARN.
+  sls_cloudformation_arn = "arn:${local.iam_partition}:cloudformation:${local.iam_region}:${local.iam_account_id}:stack/${local.sls_service_name}-${local.stage}/*"
+
+  # Serverless target deployment bucket ARN.
+  # - A long service name can endup with truncated bucket names like:
+  #   `sls-SERVICE-de-serverlessdeploymentbuck-47ati3in2360`
+  #   and possibly even more truncated, so we take a conservative approach.
+  # - No region or account id allowed. https://iam.cloudonaut.io/reference/s3.html
+  sls_deploy_bucket_arn = "arn:${local.iam_partition}:s3:::${local.sls_service_name}-*-serverless*-*"
+
+  # Serverless created log stream.
+  sls_log_stream_arn = "arn:${local.iam_partition}:logs:${local.iam_region}:${local.iam_account_id}:log-group:/aws/lambda/${local.sls_service_name}-${local.stage}-*:log-stream:"
+
+  # Serverless created CloudWatch events.
+  sls_events_arn = "arn:${local.iam_partition}:events:${local.iam_region}:${local.iam_account_id}:rule/${local.sls_service_name}-${local.stage}"
+
+  # Serverless lambda function ARN.
+  sls_lambda_arn = "arn:${local.iam_partition}:lambda:${local.iam_region}:${local.iam_account_id}:function:${local.sls_service_name}-${local.stage}-*"
+
+  # The built-in serverless Lambda execution role.
+  #
+  # _Note_: We need **actual name** to match real role, which means
+  # `local.region` and not `local.iam_region`.
+  sls_lambda_role_name = "${local.sls_service_name}-${local.stage}-${local.region}-lambdaRole"
+
+  # The built-in serverless Lambda execution role ARN.
+  #
+  # Note that we use `iam_region` to potentially wildcard the IAM permission
+  # in the actual name of the role.
+  #
+  # - No region allowed in ARN. See https://iam.cloudonaut.io/reference/iam.html
+  sls_lambda_role_arn = "arn:${local.iam_partition}:iam::${local.iam_account_id}:role/${local.sls_service_name}-${local.stage}-${local.iam_region}-lambdaRole"
+
+  # The serverless created APIGW.
+  #
+  # **NOTE**: This is difficult to lock down because we need the actual
+  # `$api-id` for a `!Ref` which we don't know ahead of time. It's
+  # created dynamically as a part of `sls` provisioning.
+  #
+  # TODO(8): Research locking down more.
+  # https://github.com/FormidableLabs/serverless-iam-terraform/issues/8
+  #
+  # See:
+  # - https://iam.cloudonaut.io/reference/apigateway.html
+  # - https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-control-access-using-iam-policies-to-invoke-api.html
```

Comment:
For future reference, you might be able to lock this down with resource policies (I've done so for a private endpoint but I think it works for any endpoint type): https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-resource-policies-create-attach.html

Reply:
Great note! Added to #8